hibernate-core/src/main/java/org/hibernate/engine/transaction/internal/TransactionImpl.java

```diff
@@ -64,8 +72,9 @@ public void begin() {
 		if ( isActive() ) {
 			throw new IllegalStateException( "Transaction already active" );
 		}
-
-		LOG.debug( "begin" );
+		if ( LOG.isDebugEnabled() ) {
```

Comment:
Why are we changing these?  Generally this is done to avoid expensive resolution of the message to log.  Here, that is not the case.  In fact, doesn't this perform worse?  IIUC this simply forces 2 "is debug enabled" checks